ambry-mysql/src/main/java/com/github/ambry/accountstats/AccountStatsMySqlStore.java

```diff
@@ -166,14 +164,23 @@ public AccountStatsMySqlStore(AccountStatsMySqlConfig config, DataSource dataSou
     partitionClassReportsDao = new PartitionClassReportsDao(dataSource, mySqlMetrics);
     this.hostnameHelper = hostnameHelper;
     storeMetrics = new AccountStatsMySqlStore.Metrics(registry);
-    if (!Strings.isNullOrEmpty(localBackupFilePath)) {
-      // load backup file and this backup is the previous stats
-      ObjectMapper objectMapper = new ObjectMapper();
+    if (config.readPreviousStatsFromMysqlDb) {
+      long startTimeMs = System.currentTimeMillis();
       try {
-        this.previousStats = objectMapper.readValue(new File(localBackupFilePath), StatsWrapper.class);
+        this.previousStats = queryAccountStatsBySimplifiedHostName(hostname, startTimeMs);
```

Comment:
why is the hostname already simplified in this situation? I may be missing something

Reply:
The hostname has already been simplified in AccountStatsMySqlStoreFactory line 59 right?